src/inmanta/agent/executor.py

```diff
@@ -320,15 +338,30 @@ async def create_environment(self, blueprint: EnvBlueprint, threadpool: ThreadPo
         :param threadpool: A ThreadPoolExecutor
         :return: An instance of ExecutorVirtualEnvironment representing the created or reused environment.
 
-        TODO: Improve handling of bad venv scenarios, such as when the folder exists but is empty or corrupted.
         """
         env_storage, is_new = self.get_or_create_env_directory(blueprint)
         process_environment = ExecutorVirtualEnvironment(env_storage, threadpool)
+        inmanta_env_status = os.path.join(env_storage, ".inmanta_env_status")
+
+        if not is_new:
+            current_status = self.read_environment_status(inmanta_env_status)
+
+            match current_status:
+                case VirtualEnvStatus.created | VirtualEnvStatus.running:
```

Comment:
Does the enum match on the string, I wonder?

Reply:
Yes, it does